### DISCUSSION TITLE
Now that Meta properties are are getting love, defining them should be made easier.

### DISCUSSION THREAD

**Discussion by SilencedPerson** [godotengine/godot-proposals#4136] [Scripting]
#58608 introduced ability to return default value when `meta` property doesn't exits.
This makes `meta` properties much more useful especially as flag list or list of properties to be serialized and saved, I would now like to be able to define them with ease much like regular variables. Syntax could go like this:
`meta name_of_meta_property = true`

**Comment by Mickeon** [comment]
Properties do not function the same way as meta properties. They're not featured in the Class definition, they are not guaranteed to exist, and their behaviour is more akin to an intrinsic Dictionary every Object *may* come with. It's more worth it make that clear with pre-defined functions that manage them. They already can be defined with ease as is.

**Reply by SilencedPerson** [reply]
All objects come with meta which is really just an empty object hence why values can be defined at runtime, it's either empty or contains some properties. I do not really see much difference with user defined variables that can also undefined. On the other hand, In this [pull request](https://github.com/godotengine/godot/pull/58608), meta properties can be accessed safely with `get_meta_or_default()`. Being able to define meta properties outside of a function is very worthwhile as it declutters functions like `_ready()` while also bringing the meta properties to forefront.

**Reply by Mickeon** [reply]
For all intents and purposes, if a variable is so important as to be needed to be defined outside of the `_ready()` method, it should just be a normal property. Splitting the concept can get confusing and brings little on the table for the average user.
